Mangata.md

```diff
@@ -0,0 +1,66 @@
+# Block construction/execution
+Mangata implements front-running bots prevention on automated market maker crypto exchange. For that reason it was decided to split block production and block execution into two following blocks. Execution of the transaction included in block N is delayed by 1 block comparing to origin substrate implementation. It is following block producer (N+1) who calculates the order of transactions execution from the previous block. This way none is able to foresee in what order transactions will be performed. 
+
+Affected creates:
+- [sc-block-builder](https://docs.rs/sc-block-builder/0.8.0/sc_block_builder/index.html) - execution of extrisnics from previous block on block creation
+
+# Shuffling
+Extrinsics are randomly shuffled but still preservers original order per every account - motivation for such algorithm was the fact that following transactions from a given account may depend on other transactions from the same account.
+
+
+Origin order:
+```
+| Alice TX1 | Alice TX2 |  Alice TX3 |  Bob TX1 | Bob TX2 |
+```
+example shuffled order:
+
+```
+| Alice TX1 |  Bob TX1  |  Alice TX2 |  Bob TX2 | Alice TX3 |
+```
+
+As shuffling occurs on block execution not creation, every node needs to calculate the same order and agree on storage state afterwards. That is achieved using Fisher-Yates shuffling algorithm with Xoshiro256++ as PRNG initialized with seed stored in blockchain storage.  Usage of sr25519 key pair together with VRF capabilities guarantees that block producer cannot impact execution order and it's easy to validate if the seed was properly calculated by validators/following nodes (thanks to VRF verification that only requires block author public key). Seed itself is stored into the storage and is publically accessible through dedicated runtime API(TODO PUT NAME HERE).
+
+PRNG initialization seed for block `N-1` is calculated as:
+- Seed from block `N-1`
+- babe epoch randomness (changed in every epoch)
```

Comment:
Isn't the randomness used in the first block of the epoch only? 
Isn't the seed encrypted?

Reply:
>Isn't the randomness used in the first block of the epoch only?
no - i wasnt aware that we want to use it in that  fashion 
> Isn't the seed encrypted?
fixed